x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/components/header_actions/search_engines_popover.tsx

```diff
@@ -15,19 +15,54 @@ import {
   EuiContextMenuPanel,
   EuiContextMenuItem,
   EuiText,
+  EuiToolTip,
 } from '@elastic/eui';
 
 import { i18n } from '@kbn/i18n';
 
 import { APP_SEARCH_PLUGIN } from '../../../../../../../common/constants';
+import { ElasticsearchIndexWithIngestion } from '../../../../../../../common/types/indices';
 import { ENGINE_CREATION_PATH } from '../../../../../app_search/routes';
+import { ESINDEX_QUERY_PARAMETER } from '../../../../../shared/constants';
 import { KibanaLogic } from '../../../../../shared/kibana';
+import { addQueryParameter } from '../../../../../shared/query_params';
+import { isHiddenIndex } from '../../../../utils/indices';
 
 import { SearchEnginesPopoverLogic } from './search_engines_popover_logic';
 
-export const SearchEnginesPopover: React.FC = () => {
+export interface SearchEnginesPopoverProps {
+  indexData?: ElasticsearchIndexWithIngestion;
+}
+
+export const SearchEnginesPopover: React.FC<SearchEnginesPopoverProps> = ({ indexData }) => {
   const { isSearchEnginesPopoverOpen } = useValues(SearchEnginesPopoverLogic);
   const { toggleSearchEnginesPopover } = useActions(SearchEnginesPopoverLogic);
+  const engineCreationURL = () => {
+    const url = `${APP_SEARCH_PLUGIN.URL}${ENGINE_CREATION_PATH}`;
+    if (!indexData?.name) {
+      return url;
+    }
+    return addQueryParameter(url, ESINDEX_QUERY_PARAMETER, indexData.name);
+  };
+  const createEngineMenuItem = (
```

Comment:
suggestion: these `const element = {}` things are an antiPattern we've been getting away from, as they mess with React's re-rendering logic. Better to embed this inside the PopOver or create a separate React.FC outside of this component.

Reply:
rgr, I was just trying to not duplicate the MenuItem when we needed to wrap it with the disabled Tooltip.